types/src/v22/wallet/error.rs

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: CC0-1.0
+
+use core::fmt;
+
+use bitcoin::address;
+
+use crate::error::write_err;
+
+/// Error when converting a `WalletDisplayAddress` type into the model type.
+#[derive(Debug)]
+pub enum WalletDisplayAddressError {
+    /// Conversion of the `address` field failed.
+    Address(address::ParseError),
+}
+
+impl fmt::Display for WalletDisplayAddressError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use WalletDisplayAddressError::*;
+        match *self {
+            Address(ref e) => write_err!(f, "conversion of the `address` field failed"; e),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for WalletDisplayAddressError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        use WalletDisplayAddressError::*;
+        match *self {
+            Address(ref e) => Some(e),
+        }
+    }
+}
```

Comment:
We don't need all this man, we can just return the `address::ParseError` directly since there is only one error. I have a vague feeling you added a single field enum recently and I missed observing this, correct?

Reply:
I have removed the error enum. I'll have a look for the other one.